libs/vkd3d/command.c

```diff
@@ -2246,6 +2250,36 @@ static uint32_t d3d12_command_list_notify_dsv_writes(struct d3d12_command_list *
     }
 }
 
+static void d3d12_command_list_notify_dsv_discard(struct d3d12_command_list *list,
+        struct d3d12_resource *resource,
+        uint32_t first_subresource, uint32_t subresource_count,
+        uint32_t resource_subresource_count)
+{
+    if (subresource_count == resource_subresource_count)
+    {
+        d3d12_command_list_promote_dsv_resource(list, resource,
+                VKD3D_DEPTH_PLANE_OPTIMAL | VKD3D_STENCIL_PLANE_OPTIMAL);
+    }
+    else if (resource->format->vk_aspect_mask == (VK_IMAGE_ASPECT_DEPTH_BIT | VK_IMAGE_ASPECT_STENCIL_BIT))
+    {
+        /* Can we at least discard a plane fully? */
+
+        if (first_subresource == 0 && subresource_count >= resource_subresource_count / 2)
```

Comment:
why `/ 2`? The subresource count that's being passed into this function does not take planes into account.

This whole thing is weird anyway since we know the d3d12_resource, so this function could just infer the subresource count from that directly.

Reply:
It looked like it did when I wrote it, but will have another look.

Reply:
Ye, it doesn't. But that's a bug I think. Gonna separate that fix out to a different PR.